src/store.cc

```diff
@@ -163,11 +163,25 @@ StoreEntry::operator delete (void *address)
     pool->freeOne(address);
 }
 
+void
+StoreEntry::makePublic()
+{
+    makePublicWith(ksDefault);
+}
+
 bool
-StoreEntry::makePublic(const KeyScope scope)
+StoreEntry::makePublicWith(const KeyScope scope)
 {
     /* This object can be cached for a long time */
-    return !EBIT_TEST(flags, RELEASE_REQUEST) && setPublicKey(scope);
+    if (EBIT_TEST(flags, RELEASE_REQUEST))
+        return false;
```

Comment:
Missing `makePrivate(true)` AFAICT -- the entry may not have been marked as valid for sharing before... And if we cannot call `makePrivate(true)` here, we need to change this method description as well!

To reduce code duplication, this method implementation may need to be refactored as 

```C++
if (!EBIT_TEST(flags, RELEASE_REQUEST) && setPublicKey(scope))
    return true;

makePrivate(true); // may already be private and shareableWhenPrivate
return false;
```